truss/templates/control/control/endpoints.py

```diff
@@ -66,47 +60,55 @@ def proxy(path):
                 ):
                     error_msg = "It appears your model has stopped running. This often means' \
                         ' it crashed and may need a fix to get it running again."
-                    return Response(error_msg, 503)
+                    return JSONResponse(error_msg, 503)
                 raise exp
 
-    headers = [(name, value) for (name, value) in resp.raw.headers.items()]
-    response = Response(resp.content, resp.status_code, headers)
+    response = Response(resp.content, resp.status_code, resp.headers)
     return response
 
 
-@control_app.route("/control/patch", methods=["POST"])
-def patch() -> Dict[str, str]:
-    current_app.logger.info("Patch request received.")
-    patch_request = request.get_json()
-    current_app.config["inference_server_controller"].apply_patch(patch_request)
-    current_app.logger.info("Patch applied successfully")
+control_app.add_route("/v1/{path:path}", proxy, ["GET", "POST"])
+
+
+@control_app.post("/control/patch")
+async def patch(request: Request) -> Dict[str, str]:
+    request.app.state.logger.info("Patch request received.")
+    patch_request = await request.json()
+    loop = asyncio.get_event_loop()
+    await loop.run_in_executor(
```

Comment:
@pankajroark @bolasim I had to make this change to be explicit about moving the patching logic to the ThreadExecutor. Without this, patching blocks the event loop, since it's very blocking.

Reply:
yeah. i had to something similar here for the background startup thread: https://github.com/basetenlabs/truss/pull/408/files#diff-6c6dd98d82346ba4d65467fddf48bcec4043cfa1cc35350520b12baa1b89f087R68

I wonder what else is blocking that we missed last time.

Reply:
The two things that the control server does is:
* apply patches & restart inference server (_is_ blockingk covered here)
* proxy stuff to inference server (non-blocking)

Anything I'm missing here?